config/sources/s500.conf

```diff
@@ -13,19 +13,18 @@ GOVERNOR=interactive
 
 case $BOARD in
 	guitar)
-	BOOTLOADER_BIN="s500-bootloader-guitar.bin"
-	UBOOT_FILES="$SRC/lib/bin/$BOOTLOADER_BIN u-boot-dtb.img"
+	UBOOT_FILES="$SRC/lib/bin/bootloader-guitar.bin u-boot-dtb.img"
 	;;
 
 	roseapple)
-	BOOTLOADER_BIN="s500-bootloader-roseapple.bin"
-	UBOOT_FILES="$SRC/lib/bin/$BOOTLOADER_BIN u-boot-dtb.img"
+	UBOOT_FILES="$SRC/lib/bin/bootloader-roseapple.bin u-boot-dtb.img"
 	;;
 esac
 
 write_uboot_platform()
 {
-	dd if=$1/$BOOTLOADER_BIN of=$2 bs=512 seek=4097 conv=fsync > /dev/null 2>&1
+	[ -f "$1/s500-bootloader-guitar.bin" ] && dd if=$1/bootloader-guitar.bin of=$2 bs=512 seek=4097 conv=fsync > /dev/null 2>&1
```

Comment:
Yes, except that I'm also using `declare -f write_uboot_platform` [here](https://github.com/igorpecovnik/lib/blob/master/common.sh#L70), which includes `write_uboot_platform` function code into postinstall script without variable expansions and substitutions, so when you decide to update u-boot, `BOOTLOADER` variable will be undefined. And because of `set -e` in the beginning of postinstall script, if first `dd` command fails, second won't be executed at all.